dataframe-arrow/src/main/kotlin/org/jetbrains/kotlinx/dataframe/io/arrowReadingImpl.kt

```diff
@@ -197,6 +235,16 @@ private fun TimeStampSecVector.values(range: IntRange): List<LocalDateTime?> =
         }
     }
 
+private fun TimeStampSecTZVector.values(range: IntRange): List<LocalDateTime?> =
+    range.mapIndexed { i, it ->
+        if (isNull(i)) {
+            null
+        } else {
+            DateUtility.getLocalDateTimeFromEpochMilli(TimeUnit.SECONDS.toMillis(getObject(it)), this.timeZone)
```

Comment:
at second glance I'm not so sure about this one.  
Parquet does not have seconds precision and my PR is for Parquet, but pyarrow feather can `.floor('S')` its datetimes.

I'm not sure what's going to be present in its .feather file, either seconds or milliseconds.

I need to test this against a `.feather` file together with seconds precision and timezone awareness, perhaps one from https://github.com/Kotlin/dataframe/tree/master/dataframe-arrow/src/test/resources

Reply:
Your pull request focuses on Parquet, but it would be beneficial to incorporate upgrades for all Arrow TimeStamp types as well, because Parquet reading (in my PR) relies totally on Arrow.
The test [testTimeStamp](https://github.com/Kotlin/dataframe/blob/54d00e69ae7c06ff254c6591242413eaf7894f01/dataframe-arrow/src/test/kotlin/org/jetbrains/kotlinx/dataframe/io/ArrowKtTest.kt#L526) covers both cases for IPC format and Feather format (by using `ArrowStreamWriter` and `ArrowFileWriter`), so you only should  add `TimeStampNanoTZVector`,`TimeStampMicroVector` and `TimeStampSecTZVector` to the [writeArrowTimestamp](https://github.com/Kotlin/dataframe/blob/54d00e69ae7c06ff254c6591242413eaf7894f01/dataframe-arrow/src/test/kotlin/org/jetbrains/kotlinx/dataframe/io/ArrowKtTest.kt#L544) method and to the expected DataFrame in `testTimeStamp` as you did for `TimeStampMicroTZVector`.
Finally as your PR improves Arrow types compatibility it could be merge independently of https://github.com/Kotlin/dataframe/pull/577 (IMHO) 😃